src/tiledb/cloud/bioimg/helpers.py

```diff
@@ -34,3 +34,89 @@ def serialize_filter(filter):
         return filter_dict
     else:
         raise TypeError
+
+
+def is_folder(path: str) -> bool:
+    return path.endswith("/")
+
+
+def validate_io_paths(source, output):
```

Comment:
This is very hard to follow. Please break it down into constituent pieces.

Also, I'm not sure why we are supporting multiple output paths. Unless there is a driving use-case, we may need to take a step back and refactor this API to take separate `source: Union[str, List[str]]` and `dest: str` arguments -- in other words, only allow a single output destination (as a separate argument). I think that will simplify this issue signficantly.

Reply:
The behaviour till now was indeed to support a single output. However by supporting multiple outputs given multiple inputs we are able to support the Quest's use-case, where the renaming of the files comes as a direct consequence. I get that the code gets complicated so I will try to simplify it further however this will probably need further restrictions on what the user can give as arguments.

Reply:
Lets move this to a design doc. We are well overdue on delivering this, I'd like to get a design agreed upon and implementation shipped, validated and accepted before end of week. Please create a document today that covers the use cases and the proposal for what we are to support.

Reply:
https://docs.google.com/document/d/1QDIF0T1mWsaOVswsARy5NKGHXc-Vn98083Sv5dWt6MM/edit?usp=sharing